.github/workflows/testing.yml

```diff
@@ -0,0 +1,210 @@
+        msgpack-deps:
+          # latest msgpack will be installed as a part of requirements.txt
+          - ''
+
+        # Adding too many elements to three-dimentional matrix results in
+        # too many test cases. It causes GitHub webpages to fail with 
+        # "This page is taking too long to load." error. Thus we use
+        # pairwise testing. 
+        include:
+          - tarantool: '2.8'
+            python: '3.10'
+            msgpack-deps: 'msgpack-python==0.4.0'
+          - tarantool: '2.8'
+            python: '3.10'
+            msgpack-deps: 'msgpack==0.5.0'
+          - tarantool: '2.8'
+            python: '3.10'
+            msgpack-deps: 'msgpack==0.6.2'
+          - tarantool: '2.8'
+            python: '3.10'
+            msgpack-deps: 'msgpack==1.0.0'
+
+    steps:
+      - name: Clone the connector
+        uses: actions/checkout@v2
+
+      - name: Install tarantool ${{ matrix.tarantool }}
+        if: matrix.tarantool != '2.x-latest'
+        uses: tarantool/setup-tarantool@v1
+        with:
+          tarantool-version: ${{ matrix.tarantool }}
+
+      - name: Install latest tarantool 2.x
+        if: matrix.tarantool == '2.x-latest'
+        run: |
+          curl -L https://tarantool.io/pre-release/2/installer.sh | sudo bash
+          sudo apt install -y tarantool tarantool-dev
+
+      - name: Setup Python for tests
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+
+      - name: Install msgpack python package
+        if: matrix.msgpack-deps != ''
+        run: pip install ${{ matrix.msgpack-deps }}
+
+      - name: Install connector requirements
+        # msgpack package is a replacement for deprecated msgpack-python.
+        # To test compatibility with msgpack-python we must ignore
+        # requirements.txt install, since it will install newest msgpack.
+        # Beware that if any new dependency will be added to requirements.txt,
+        # this step will be invalid.
+        if: matrix.msgpack-deps != 'msgpack-python==0.4.0'
+        run: pip install -r requirements.txt
+
+      - name: Install test requirements
+        run: pip install -r requirements-test.txt
+
+      - name: Run tests
+        run: make test
+
+  run_tests_windows:
+    # We want to run on external PRs, but not on our own internal
+    # PRs as they'll be run by the push to the branch.
+    #
+    # The main trick is described here:
+    # https://github.com/Dart-Code/Dart-Code/pull/2375
+    if: github.event_name == 'push' ||
+      github.event.pull_request.head.repo.full_name != github.repository
+
+    runs-on: windows-2022
+
+    strategy:
+      fail-fast: false
+      matrix:
+        tarantool:
+          - '1.10'
+          - '2.8'
+          - '2.9'
+          - '2.x-latest'
+        python:
+          - '2.7'
+          - '3.4'
+          - '3.5'
+          - '3.6'
+          - '3.7'
+          - '3.8'
+          - '3.9'
+          - '3.10'
```

Comment:
The workflows on Windows are not cheap. Maybe keep [tarantool-1.10, tarantool-latest] × [python-2.7, python-3.10] plus different msgpack versions?

Reply:
Decided to leave only [tarantool-1.10, tarantool-2.8] × [python-2.7, python-3.10] since tarantool-latest is not stable and different msgpack version testing on Windows may be a bit too much.